### PR TITLE
Suppress SecurityCritical attributes

### PR DESCRIPTION
**Pull request by kevingosse**
Contributes to https://github.com/dotnet/winforms/issues/48 

- Restore compiler warning CS0649: there actually no unused field in this project
- Fix warnings due to SecurityCriticalScope being obsolete

### CODE REVIEW

**Comment by kevingosse** [comment]
This fixes about 12 warnings, 201 to go. Most of them seem to be about marshalling.

File: src/System.Windows.Forms/src/System/Windows/Forms/UnsafeNativeMethods.cs [hunk 1]
```diff
@@ -7400,7 +7400,7 @@ public sealed class tagQACONTROL
 
     }
 
-    [SecurityCritical(SecurityCriticalScope.Everything)]
+    [SecurityCritical]
```
Comment (karelz) [inline src/System.Windows.Forms/src/System/Windows/Forms/UnsafeNativeMethods.cs]: Do we need the security attributes at all? CoreFX does not have any such attributes anymore: https://github.com/dotnet/corefx/search?q=SecurityCritical&unscoped_q=SecurityCritical

@ericstj can you please advise?

**Comment by kevingosse** [comment]
I reenabled CS0649. It looks like there's actually two occurrences of those warnings, I missed them because they only happen in RELEASE mode. Started a discussion here on how to deal with them: https://github.com/dotnet/winforms/issues/48#issuecomment-445296374

File: src/System.Windows.Forms/src/System/Windows/Forms/UnsafeNativeMethods.cs (hunk 1 above)
Comment (ericstj) [reply src/System.Windows.Forms/src/System/Windows/Forms/UnsafeNativeMethods.cs]: Right, all CAS should be removed as it is a noop on .NETCore.  There was an internal issue for this that likely hasn't been copied to winforms/wpf repos.

Comment (kevingosse) [reply src/System.Windows.Forms/src/System/Windows/Forms/UnsafeNativeMethods.cs]: I'm going to suppress them globally then

Comment (ericstj) [reply src/System.Windows.Forms/src/System/Windows/Forms/UnsafeNativeMethods.cs]: Actually, here's an issue: https://github.com/dotnet/winforms/issues/40

**Pull request merged**